core/src/color.rs

```diff
@@ -80,16 +93,123 @@ impl Color {
             self.a,
         ]
     }
+
+    /// Invert the Color in-place
+    pub fn invert(&mut self) {
+        self.r = clamp(1.0f32 - self.r);
+        self.b = clamp(1.0f32 - self.g);
+        self.g = clamp(1.0f32 - self.b);
+    }
+
+    /// Return an inverted Color
+    pub fn inverse(self) -> Color {
+        Color::new(1.0f32 - self.r, 1.0f32 - self.g, 1.0f32 - self.b, self.a)
+    }
 }
 
 impl From<[f32; 3]> for Color {
     fn from([r, g, b]: [f32; 3]) -> Self {
-        Color { r, g, b, a: 1.0 }
+        Color::new(r, g, b, 1.0)
     }
 }
 
 impl From<[f32; 4]> for Color {
     fn from([r, g, b, a]: [f32; 4]) -> Self {
-        Color { r, g, b, a }
+        Color::new(r, g, b, a)
+    }
+}
+
+impl From<HSLColor> for Color {
+    fn from(hsl: HSLColor) -> Self {
+        // Compute Chroma
+        let ch = (1.0 - (2.0 * hsl.l - 1.0).abs()) * hsl.s;
+
+        // Quantized Hue: H'
+        let hp: u8 = (hsl.h / 60.0).ceil() as u8;
+        let x: f32 = ch * f32::from(1 - ((hp % 2) - 1));
+
+        // Intermediate RGB values
+        let (r1, g1, b1): (f32, f32, f32) = match hp {
+            1 => (ch, x, 0.0),
+            2 => (x, ch, 0.0),
+            3 => (0.0, ch, x),
+            4 => (0.0, x, ch),
+            5 => (x, 0.0, ch),
+            6 => (ch, 0.0, x),
+            _ => (0.0, 0.0, 0.0),
+        };
+
+        // Match lightness
+        let m = hsl.l - ch / 2.0;
+
+        Color::new(r1 + m, g1 + m, b1 + m, hsl.a)
+    }
+}
+
+/// A color in the HSL color space.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct HSLColor {
+    /// Hue, 0.0 - 360.0
+    pub h: f32,
+    /// Saturation, 0.0 - 1.0
+    pub s: f32,
+    /// Lightness, 0.0 - 1.0
+    pub l: f32,
+    /// Transparency, 0.0 - 1.0
+    pub a: f32,
+}
```

Comment:
Here we need to decide how much of color conversion logic we want to include in the library.

My idea was to offer a simple sRGB `Color` type as the library boundary and let users choose freely how to deal with color conversions ([`palette`](https://docs.rs/palette/0.5.0/palette/index.html) is pretty nice). Maybe we could add a feature gate to automatically implement conversions for `Color` from `palette`'s types.

Reply:
Ooh, `palette` looks nice! I didn't do too much searching for existing color libraries, but I assumed there was something out there.

Feature gate to bring in the much more advanced features is a way better approach than re-implementation ;-)

Two-way conversion would be very cool. Perhaps I'll change the direction of this branch to look into that idea.

Reply:
Tie-in to palette's `Srgba` in 59f6fa4718d8e416bf137967994843f2426e8ad4